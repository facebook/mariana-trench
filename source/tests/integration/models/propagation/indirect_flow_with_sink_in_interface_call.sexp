(abstract method (public) "LInterface;.sink:(LData;)V"
 (
   LINE(1)
   (return-void)
 )
)
(super "LInterface;")
(method (public) "LImplementation;.sink:(LData;)V"
 (
  (load-param v0)
  (load-param v1)

  LINE(1)
  (invoke-virtual (v0 v1) "LSink;.sink:(LData;)V")
  (return-void)
 )
)
(method (public) "LFlow;.flow:()V"
 (
  LINE(1)
  (invoke-virtual (v0) "LSource;.source:()LData;")
  (move-result-object v0)

  LINE(2)
  (invoke-virtual (v2 v0) "LInterface;.sink:(LData;)V")
  (return-void)
 )
)
