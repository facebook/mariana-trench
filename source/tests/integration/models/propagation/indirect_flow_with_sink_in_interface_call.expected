{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "issues" :
      [
        {
          "position" :
          {
            "line" : 2,
            "path" : "Flow.java"
          },
          "rule" : 1,
          "sinks" :
          [
            {
              "call_position" :
              {
                "line" : 2,
                "path" : "Flow.java"
              },
              "callee" : "LImplementation;.sink:(LData;)V",
              "callee_port" : "Argument(1)",
              "distance" : 2,
              "kind" : "TestSink",
              "origins" :
              [
                "LSink;.sink:(LData;)V"
              ]
            }
          ],
          "sources" :
          [
            {
              "call_position" :
              {
                "line" : 1,
                "path" : "Flow.java"
              },
              "callee" : "LSource;.source:()LData;",
              "callee_port" : "Return",
              "distance" : 1,
              "kind" : "TestSource",
              "origins" :
              [
                "LSource;.source:()LData;"
              ]
            }
          ]
        }
      ],
      "method" : "LFlow;.flow:()V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      }
    },
    {
      "method" : "LImplementation;.sink:(LData;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Implementation.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Implementation.java"
          },
          "callee" : "LSink;.sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(1)",
          "distance" : 1,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    },
    {
      "method" : "LInterface;.sink:(LData;)V",
      "modes" :
      [
        "add-via-obscure-feature",
        "skip-analysis",
        "taint-in-taint-out",
        "taint-in-taint-this"
      ],
      "position" :
      {
        "path" : "Interface.java"
      },
      "propagation" :
      [
        {
          "always_features" :
          [
            "via-obscure",
            "via-obscure-taint-in-taint-this"
          ],
          "input" : "Argument(1)",
          "output" : "Argument(0)"
        }
      ]
    }
  ]
}
